src/sequence_state.cc

```diff
@@ -85,7 +90,13 @@ SequenceState::SetStringDataToZero()
   const std::shared_ptr<MutableMemory>& memory =
       reinterpret_cast<const std::shared_ptr<MutableMemory>&>(Data());
   char* buffer = memory->MutableBuffer(&memory_type, &memory_type_id);
-  memset(buffer, 0, Data()->TotalByteSize());
+  if (memory_type == TRITONSERVER_MEMORY_GPU) {
+    RETURN_IF_CUDA_ERR(
+        cudaMemset(buffer, 0, Data()->TotalByteSize()),
+        std::string("failed to set the data to zero."));
+  } else {
+    memset(buffer, 0, Data()->TotalByteSize());
```

Comment:
It almost seems like ClearMemory / SetMemory should be methods on the memory object and it would encapsulate the difference for different memory types.

It's odd for the sequence state to have to know details about the type of memory?

Reply:
It is not the details/differences of memory type. The reason that we have this in SequenceState is that this special condition is only required for SequenceState with String datatype. By default for initial state we can have garbage data which is not acceptable for string data type (since the first element is the number of bytes which means only certain values are allowed for the rest of the buffer).

I see your point regarding having utilities in Memory to clear the memory though and I can make that change.

Reply:
fixed.